source/renderer/app/components/wallet/settings/WalletSettings.js

```diff
@@ -172,6 +225,90 @@ export default class WalletSettings extends Component<Props, State> {
     );
   };
 
+  onUndelegateWalletClick = async () => {
+    const {
+      walletId,
+      openDialogAction,
+      updateDataForActiveDialogAction,
+    } = this.props;
+    this.onBlockForm();
+    openDialogAction({
+      dialog: UndelegateWalletConfirmationDialog,
+    });
+    updateDataForActiveDialogAction({
+      data: { walletId },
+    });
+  };
+
+  renderUndelegateWalletBox = () => {
+    const { intl } = this.context;
+    const {
+      lastDelegationStakePoolStatus,
+      isRestoring,
+      isSyncing,
+      isLegacy,
+      walletReward,
+      isDialogOpen,
+      onDelegateClick,
+      undelegateWalletDialogContainer,
+    } = this.props;
+    const isDelegating =
```

Comment:
@yakovkaravelov this is not good. We need to track active status too. Thank you Yakov!